### PR TITLE
feat: Allow MariaDB cluster creation in multiple locations

### PR DESCRIPTION
**Pull request by adeatcu-ionos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/resource_dbaas_mariadb_cluster.go [hunk 1]
```diff
@@ -171,7 +179,15 @@ func mariaDBClusterCreate(ctx context.Context, d *schema.ResourceData, meta inte
 func mariaDBClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(services.SdkBundle).MariaDBClient
 	clusterID := d.Id()
-	_, apiResponse, err := client.DeleteCluster(ctx, d.Id())
+	var location string
+	// This takes care of the old version of the provider where the location was not required and
+	// all the requests were done for "de/txl" location with the correspondent endpoint.
+	if locationValue, locationOk := d.GetOk("location"); locationOk {
```
Comment (cbeti-ionos) [inline ionoscloud/resource_dbaas_mariadb_cluster.go:185]: doesn't `location` being Required mean `GetOk` is redundant here ?

Comment (adeatcu-ionos) [reply ionoscloud/resource_dbaas_mariadb_cluster.go:185]: @cbeti-ionos no, this takes care of the case in which the user has an older version of the provider that doesn't have the `location` attribute.

File: ionoscloud/resource_dbaas_mariadb_cluster.go [hunk 2]
```diff
@@ -192,19 +208,31 @@ func mariaDBClusterDelete(ctx context.Context, d *schema.ResourceData, meta inte
 
 func mariaDBClusterImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
 	client := meta.(services.SdkBundle).MariaDBClient
-	clusterId := d.Id()
-	cluster, apiResponse, err := client.GetCluster(ctx, clusterId)
+	// TODO -- Change the documentation for import
```
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/resource_dbaas_mariadb_cluster.go]: @cbeti-ionos done

**Comment by adeatcu-ionos** [comment]
@cbeti-ionos I made the `location` attribute optional, please take another look when you get the chance. Thank you!

**Review by cbeti-ionos — APPROVED** [review]

**Pull request merged**